internal/resolution/pm/pip/job.go

```diff
@@ -65,25 +66,59 @@ func (j *Job) Run() {
 			if j.venvPath == "" {
 				return
 			}
-			j.SendStatus("removing venv")
+			status := "removing venv"
+			j.SendStatus(status)
 			err := j.pipCleaner.RemoveAll(j.venvPath)
 			if err != nil {
-				j.Errors().Critical(err)
+				cmdErr := util.NewPMJobError(err.Error())
+				cmdErr.SetDocumentation("Error when trying to remove previous virtual environment")
+				cmdErr.SetStatus(status)
+				j.Errors().Critical(cmdErr)
 			}
 		}()
-
-		j.SendStatus("creating venv")
-		_, err := j.runCreateVenvCmd()
-		if err != nil {
-			j.Errors().Critical(err)
+		status := "creating venv"
+		j.SendStatus(status)
+		_, cmdErr := j.runCreateVenvCmd()
+		if cmdErr != nil {
+			cmdErr.SetDocumentation("Error when trying to create python virtual environment")
+			cmdErr.SetStatus(status)
+			j.Errors().Critical(cmdErr)
 
 			return
 		}
-
-		j.SendStatus("installing requirements")
-		_, err = j.runInstallCmd()
-		if err != nil {
-			j.Errors().Critical(err)
+		status = "installing dependencies"
+		j.SendStatus(status)
+		_, cmdErr = j.runInstallCmd()
+		if cmdErr != nil {
+			cmdErr.SetStatus(status)
+			var buildError = regexp.MustCompile("python setup.py ([^ ]*) did not run successfully.")
+			var credentialError = regexp.MustCompile("WARNING: 401 Error, Credentials not correct for")
+
+			switch {
+			case buildError.MatchString(cmdErr.Error()):
+				dependencyNameMatch := buildError.FindStringSubmatch(cmdErr.Error())
+				dependencyName := ""
+				if len(dependencyNameMatch) > 1 {
+					dependencyName = "\"" + dependencyNameMatch[1] + "\""
+				}
+				cmdErr.SetDocumentation(
+					strings.Join(
+						[]string{
+							"Failed to build python dependecy ",
+							dependencyName,
+							"with setup.py, this probably means the " +
+								"project was not set up correctly and " +
+								"could mean that a OS package is missing.",
+						}, ""),
+				)
+			case credentialError.MatchString(cmdErr.Error()):
+				cmdErr.SetDocumentation(
+					"Could not install dependencies due to authorization, " +
+						"this could mean you have private dependencies which the" +
+						" debricked CLI does not have access to.",
```

Comment:
```suggestion
					"Could not install dependencies due to authorization. " +
						"This could mean you have private dependencies that the" +
						" debricked CLI does not have access to.",
```